openmdao/recorders/dump_recorder.py

```diff
@@ -90,17 +89,51 @@ def record(self, params, unknowns, resids, metadata):
         fmat = "Iteration Coordinate: {0:s}\n"
         write(fmat.format(format_iteration_coordinate(iteration_coordinate)))
 
-        write("Params:\n")
-        for param, val in sorted(iteritems(params)):
-            write("  {0}: {1}\n".format(param, str(val)))
+        if self.options['record_params']:
+            write("Params:\n")
+            for param, val in sorted(iteritems(params)):
+                write("  {0}: {1}\n".format(param, str(val)))
 
-        write("Unknowns:\n")
-        for unknown, val in sorted(iteritems(unknowns)):
-            write("  {0}: {1}\n".format(unknown, str(val)))
+        if self.options['record_unknowns']:
+            write("Unknowns:\n")
+            for unknown, val in sorted(iteritems(unknowns)):
+                write("  {0}: {1}\n".format(unknown, str(val)))
 
-        write("Resids:\n")
-        for resid, val in sorted(iteritems(resids)):
-            write("  {0}: {1}\n".format(resid, str(val)))
+        if self.options['record_resids']:
+            write("Resids:\n")
+            for resid, val in sorted(iteritems(resids)):
+                write("  {0}: {1}\n".format(resid, str(val)))
 
         # Flush once per iteration to allow external scripts to process the data.
         self.out.flush()
+
+    def record_metadata(self, group):
+        """Dump the metadata of the given group in a "pretty" form.
+
+        Args
+        ----
+        group : `System`
+            `System` containing vectors 
+        """
+        params = list(iteritems(group.params))
+        unknowns = list(iteritems(group.unknowns))
+        resids = list(iteritems(group.resids))
+
+        self.out.write("Metadata:\n")
+        self.out.write("Params:\n")
+
+        for name, metadata in params:
+            fmat = "  {0}: {1}\n"
+            self.out.write(fmat.format(name, metadata))
+        
+        self.out.write("Unknowns:\n")
+
+        for name, metadata in unknowns:
+            fmat = "  {0}: {1}\n"
+            self.out.write(fmat.format(name, metadata))
+        
+        self.out.write("Resids:\n")
```

Comment:
Unless I hear otherwise from someone else by the end of the day, I will remove the recording of the resids metadata from the dump, hdf5, and sqlite recorders.